ucan.ipldsch

```diff
@@ -34,11 +34,47 @@ type Resource = String
 -- Must be all lower-case `/` delimeted with at least one path segment
 type Ability = String
 
+
+type Principal union {
+  -- Specification defines principals in terms of did:key's (that are multicodec
+  -- identifier for the public key type followed by the raw bytes of the key)
+  -- we represent those as raw public key bytes prefixed with public key multiformat
+  -- code.
+  | Ed25519    "0xed"
+  | RSA        "0x1205"
+  | P256       "0x1200"
+  | P384       "0x1201"
+  | P521       "0x1202"
```

Comment:
Do we want to register specific ones, or is there a way to say basically "any key in the multicodec table" (e.g. BLS, which Fission uses for a few things). Here's what's in the did:key spec at least:

<img width="239" alt="Screen Shot 2022-09-14 at 14 07 30" src="https://user-images.githubusercontent.com/1052016/190262641-4b3c0ed6-3042-45d9-84f9-41bf3b383605.png">

Reply:
Oh yeah we also need secp256k1 for sure

Reply:
They would have to be specific ones in the union definition. However spec could say any public key in multicodec table is good.